runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java

```diff
@@ -165,20 +164,26 @@ private MonitoringInfo counterUpdateToMonitoringInfo(MetricUpdate<Long> metricUp
       for (Entry<String, String> e : monitoringInfoName.getLabels().entrySet()) {
         builder.setLabel(e.getKey(), e.getValue());
       }
-    } else { // Note: (metricName instanceof MetricName) is always True.
-      // Represents a user counter.
-      builder.setUrnForUserMetric(
-          metricUpdate.getKey().metricName().getNamespace(),
-          metricUpdate.getKey().metricName().getName());
+    } else { // Represents a user counter.
       // Drop if the stepname is not set. All user counters must be
       // defined for a PTransform. They must be defined on a container bound to a step.
       if (this.stepName == null) {
         return null;
```

Comment:
Should we log that the counter is being dropped? So we can figure out what user counters we're not appending the stepName for?

Reply:
Valid comment, but I suggest to keep it as-is for this PR.
My assumption is that we do not do that due to potential spam in logs.